### PR TITLE
Document all items in `query_dsl`.

### PR DESCRIPTION
**Pull request by sgrif**

### CODE REVIEW

File: diesel/src/query_dsl/join_dsl.rs [hunk 1]
```diff
@@ -44,34 +44,35 @@ where
...
+/// # #[macro_use] extern crate diesel;
+/// # include!("../doctest_setup.rs");
+/// # use schema::{users, posts};
+/// #
+/// # fn main() {
+/// #     let connection = establish_connection();
+/// let data = users::table
+///     .left_join(posts::table.on(
+///         users::id.eq(posts::user_id).and(
+///             posts::title.eq("My first post"))
```
Comment (killercup) [inline diesel/src/query_dsl/join_dsl.rs]: Btw, I'd format this as

```rust
users::id.eq(posts::user_id)
.and(posts::title.eq("My first post"))
```

File: diesel/src/query_dsl/save_changes_dsl.rs [hunk 2]
```diff
@@ -66,45 +66,46 @@ where
...
+/// #    legs: i32,
+/// #    name: Option<String>,
+/// # }
+/// #
+/// # fn main() {
+/// #     use animals::dsl::*;
+/// #     let connection = establish_connection();
+/// let mut spider = animals.filter(species.eq("spider"))
+///     .first::<Animal>(&connection)
+///     .expect("Too scary to load");
```
Comment (killercup) [inline diesel/src/query_dsl/save_changes_dsl.rs]: maybe change this test to use a wrapper fn and `?` too?

File: diesel/src/query_dsl/join_dsl.rs (hunk 1 above)
Comment (sgrif) [reply diesel/src/query_dsl/join_dsl.rs]: This is just moved from the other location. I wonder if we can set up a way to run our docs code through rustfmt.

File: diesel/src/query_dsl/save_changes_dsl.rs (hunk 2 above)
Comment (sgrif) [reply diesel/src/query_dsl/save_changes_dsl.rs]: This is just moved from its original location, and I also have a PR up that rewrites this example

**Pull request merged**